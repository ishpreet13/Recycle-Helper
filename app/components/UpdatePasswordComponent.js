import React from "react";
import { StyleSheet, View, TouchableOpacity } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import colors from "../config/colors";
import { TextInput } from "react-native-gesture-handler";

function UpdatePasswordComponent({
  leftIcon,
  rightIcon,
  placeholder,
  handlePasswordVisibility,
  ...rest
}) {
  return (
    <View style={styles.container}>
      {leftIcon ? (
        <MaterialCommunityIcons
          name={leftIcon}
          size={20}
          style={styles.leftIcon}
        />
      ) : null}
      <TextInput
        {...rest}
        placeholder={placeholder}
        placeholderTextColor="grey"
        style={styles.input}
      ></TextInput>
      {rightIcon ? (
        <TouchableOpacity onPress={handlePasswordVisibility}>
          <MaterialCommunityIcons
            name={rightIcon}
            size={20}
            style={styles.rightIcon}
          />
        </TouchableOpacity>
      ) : null}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.white,
    borderRadius: 10,
    flexDirection: "row",
    alignItems: "center",
    margin: 10,
    marginBottom: 10,
    borderWidth: 1,
    borderStyle: "solid",
    borderColor: colors.black,
  },
  input: {
    flex: 1,
    backgroundColor: colors.white,
    height: 22,
    margin: 10,
    textAlignVertical: "center",
    borderRadius: 10,
    color: "grey",
    fontSize: 15,
  },
  leftIcon: {
    marginRight: 10,
    color: colors.black,
  },
  rightIcon: {
    marginRight: 10,
    alignSelf: "center",
  },
});

export default UpdatePasswordComponent;
