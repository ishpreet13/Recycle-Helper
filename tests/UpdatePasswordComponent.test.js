import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import UpdatePasswordComponent from "../app/components/UpdatePasswordComponent";

describe("Testing UpdatePasswordComponent ", () => {
  const wrapper = renderer.create(<UpdatePasswordComponent />);
  it("should render", () => {
    expect(wrapper.toJSON()).toBeTruthy();
  });
});

describe("<UpdatePasswordComponent />", () => {
  let tree;
  it("should render correctly", async () => {
    tree = renderer.create(<UpdatePasswordComponent />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
