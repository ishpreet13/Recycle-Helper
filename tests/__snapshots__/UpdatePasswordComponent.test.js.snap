// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdatePasswordComponent /> should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#000",
      "borderRadius": 10,
      "borderStyle": "solid",
      "borderWidth": 1,
      "flexDirection": "row",
      "margin": 10,
      "marginBottom": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    placeholderTextColor="grey"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 10,
        "color": "grey",
        "flex": 1,
        "fontSize": 15,
        "height": 22,
        "margin": 10,
        "textAlignVertical": "center",
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
